### ISSUE TITLE
The `s3` backend is not very suitable for common use cases

### ISSUE THREAD

**Issue by DavidGregory084** [buildkite-plugins/cache-buildkite-plugin#46]
I've tried using the S3 backend with a configuration like the one from the README:

```diff
steps:
  - label: ':nodejs: Install dependencies'
    command: npm ci
    plugins:
      - cache#v0.4.0:
+          backend: s3
           manifest: package-lock.json
           path: node_modules
           restore: file
           save: file
```

There are problems with using S3 to cache node_modules, especially via the AWS CLI:

1. S3 does not preserve file permissions
1. S3 does not preserve symlinks

These problems are pretty noticeable for TypeScript projects, since `tsc` is no longer executable once it is restored from S3:

<img width="229" alt="image" src="https://github.com/buildkite-plugins/cache-buildkite-plugin/assets/2992938/1aa93399-c6b6-45ff-a86d-6d4250c411e7">

If we manage to fix that problem (e.g. by using MinIO client instead of AWS CLI to restore the file permissions), we will run into the next problem - restoring `node_modules` from S3 replaces all of the symlinks in `node_modules/.bin` with a duplicate file containing the contents of the symlink target, which leads to obscure errors when running executables:

<img width="686" alt="image" src="https://github.com/buildkite-plugins/cache-buildkite-plugin/assets/2992938/499589f1-c0c8-418d-91e0-e7f5715118da">

I don't know if other build tools rely quite so heavily on symlinks, but it seems unwise to advertise a workflow that is unlikely to work for a lot of folks so prominently in the README.

Perhaps these limitations should be documented until something like #44 is merged?

**Comment by pzeballos** [comment]
Hey @DavidGregory084! Sorry for the delay 🙏🏻! #44 is already merged and part of the [v0.5.0](https://github.com/buildkite-plugins/cache-buildkite-plugin/releases/tag/v0.5.0) release, where you able to test it out and see if that takes care of that?

**Comment by DavidGregory084** [comment]
Thanks @pzeballos! I've tried out #44 to see whether it resolves my issue but unfortunately I ran into #52 while I was trying it out, so I'm not sure whether it resolves these issues yet.

**Comment by DavidGregory084** [comment]
I've tried out a fix for #52 and everything otherwise seems to work well, so let's close this!
